internal/promotion/runner/builtin/schemas/git-clone-config.json

```diff
@@ -79,6 +79,18 @@
           }
         ]
       }
+    },
+    "user": {
+      "type": "string",
+      "description": "The name of the Git user."
+    },
+    "email": {
+      "type": "string",
+      "description": "The email of the Git user."
+    },
+    "signingKey": {
+      "type": "string",
+      "description": "The GPG signing key to use for signing commits."
```

Comment:
It would make more sense for `name`, `email`, and `signingKey` to be attributes of a `user` field. i.e.: config should resemble this:

```yaml
- uses: git-clone
  config:
    repoURL: ${{ vars.gitRepo }}
    user:
      name: Tony Stark
      email: tony@starkindustries.io
      signingKey: ${{ secret('my-key').gpg }}
    checkout:
    - commit: ${{ commitFrom(vars.gitRepo).ID }}
      path: ./src
    - branch: stage/${{ ctx.stage }}
      create: true
      path: ./out
```

Reply:
`user` should be an optional field. If non-empty, `name` and `email` should be required, with `signingKey` optional.